nix/tahoe-lafs.nix

```diff
@@ -0,0 +1,57 @@
+{ fetchFromGitHub, lib
+, nettools, python
+, twisted, foolscap, nevow, zfec
+, setuptools, setuptoolsTrial, pyasn1, zope_interface
+, service-identity, pyyaml, magic-wormhole, treq, appdirs
+, beautifulsoup4, eliot, autobahn, cryptography
+, html5lib
+}:
+python.pkgs.buildPythonPackage rec {
+  version = "1.14.0.dev";
+  name = "tahoe-lafs-${version}";
+  src = lib.cleanSource ../.;
+
+  postPatch = ''
+    sed -i "src/allmydata/util/iputil.py" \
+        -es"|_linux_path = '/sbin/ifconfig'|_linux_path = '${nettools}/bin/ifconfig'|g"
+
+    # Chroots don't have /etc/hosts and /etc/resolv.conf, so work around
+    # that.
+    for i in $(find src/allmydata/test -type f)
+    do
+      sed -i "$i" -e"s/localhost/127.0.0.1/g"
+    done
+
+    # Some tests are flaky or fail to skip when dependencies are missing.
```

Comment:
Would be nice to note which dependencies the Nix environment is not installing.

Reply:
Indeed.  Added some comments.  And hopefully we shrink this list to nothing eventually.